bindings/pyroot/pythonizations/python/ROOT/_pythonization/_tmva/_batchgenerator.py

```diff
@@ -306,18 +311,21 @@ def ConvertBatchToNumpy(self, batch: "RTensor") -> np.ndarray:
             raise ImportError("Failed to import numpy in batchgenerator init")
 
         data = batch.GetData()
-        data.reshape((self.batch_size * self.num_columns,))
 
-        return_data = np.array(data).reshape(self.batch_size, self.num_columns)
+        if tuple(batch.GetShape()) != (self.batch_size,self.num_columns):
+            batch_size, num_columns = tuple(batch.GetShape())
+        else:
+            batch_size = self.batch_size
+            num_columns = self.num_columns
```

Comment:
Why is this done unconditionally? What is a situation where the first `if` condition might be true? e.g. when is the shape of the batch different from the stored attributes? This practically means that what the user asked initially is not what they will get

Reply:
Condition check will be removed